[2025-03-06T16:03:46.874+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-06T16:03:46.905+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:03:42.400789+00:00 [queued]>
[2025-03-06T16:03:46.919+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:03:42.400789+00:00 [queued]>
[2025-03-06T16:03:46.920+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-03-06T16:03:46.936+0000] {taskinstance.py:2890} INFO - Executing <Task(SparkSubmitOperator): load_exchangeRate_python_job> on 2025-03-06 16:03:42.400789+00:00
[2025-03-06T16:03:46.944+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=3436) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-03-06T16:03:46.945+0000] {standard_task_runner.py:72} INFO - Started process 3462 to run task
[2025-03-06T16:03:46.944+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_flow', 'load_exchangeRate_python_job', 'manual__2025-03-06T16:03:42.400789+00:00', '--job-id', '49', '--raw', '--subdir', 'DAGS_FOLDER/dag_***_spark.py', '--cfg-path', '/tmp/tmpkqe2aonf']
[2025-03-06T16:03:46.946+0000] {standard_task_runner.py:105} INFO - Job 49: Subtask load_exchangeRate_python_job
[2025-03-06T16:03:47.012+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:03:42.400789+00:00 [running]> on host e9dfa152b6ab
[2025-03-06T16:03:47.127+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='haonv' AIRFLOW_CTX_DAG_ID='spark_flow' AIRFLOW_CTX_TASK_ID='load_exchangeRate_python_job' AIRFLOW_CTX_EXECUTION_DATE='2025-03-06T16:03:42.400789+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-06T16:03:42.400789+00:00'
[2025-03-06T16:03:47.128+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-03-06T16:03:47.167+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2025-03-06T16:03:47.169+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://172.18.0.4:7077 --conf spark.jars=file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py
[2025-03-06T16:03:50.751+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:50 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-03-06T16:03:50.967+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:50 WARN DependencyUtils: Local jar /opt/bitnami/spark/jars/postgresql-42.7.5.jar does not exist, skipping.
[2025-03-06T16:03:52.391+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SparkContext: Running Spark version 3.5.5
[2025-03-06T16:03:52.392+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SparkContext: OS info Linux, 5.15.167.4-microsoft-standard-WSL2, amd64
[2025-03-06T16:03:52.392+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SparkContext: Java version 17.0.14
[2025-03-06T16:03:52.415+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO ResourceUtils: ==============================================================
[2025-03-06T16:03:52.416+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-03-06T16:03:52.416+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO ResourceUtils: ==============================================================
[2025-03-06T16:03:52.417+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SparkContext: Submitted application: ExchangeRateToDW
[2025-03-06T16:03:52.433+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-03-06T16:03:52.440+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO ResourceProfile: Limiting resource is cpu
[2025-03-06T16:03:52.441+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-03-06T16:03:52.479+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SecurityManager: Changing view acls to: ***
[2025-03-06T16:03:52.480+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SecurityManager: Changing modify acls to: ***
[2025-03-06T16:03:52.481+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SecurityManager: Changing view acls groups to:
[2025-03-06T16:03:52.481+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SecurityManager: Changing modify acls groups to:
[2025-03-06T16:03:52.482+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2025-03-06T16:03:52.673+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO Utils: Successfully started service 'sparkDriver' on port 39391.
[2025-03-06T16:03:52.695+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SparkEnv: Registering MapOutputTracker
[2025-03-06T16:03:52.726+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SparkEnv: Registering BlockManagerMaster
[2025-03-06T16:03:52.739+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-03-06T16:03:52.740+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-03-06T16:03:52.743+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-03-06T16:03:52.771+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-995faf95-54b9-40a2-85c4-abcf36e61eac
[2025-03-06T16:03:52.782+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-03-06T16:03:52.799+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-03-06T16:03:52.913+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-03-06T16:03:52.975+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:52 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-03-06T16:03:53.012+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 ERROR SparkContext: Failed to add file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar to Spark environment
[2025-03-06T16:03:53.013+0000] {spark_submit.py:641} INFO - java.io.FileNotFoundException: Jar /opt/bitnami/spark/jars/postgresql-42.7.5.jar not found
[2025-03-06T16:03:53.013+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:2095)
[2025-03-06T16:03:53.014+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2151)
[2025-03-06T16:03:53.015+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15(SparkContext.scala:521)
[2025-03-06T16:03:53.016+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15$adapted(SparkContext.scala:521)
[2025-03-06T16:03:53.016+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-03-06T16:03:53.017+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-03-06T16:03:53.018+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-03-06T16:03:53.018+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:521)
[2025-03-06T16:03:53.019+0000] {spark_submit.py:641} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-03-06T16:03:53.020+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-03-06T16:03:53.020+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-03-06T16:03:53.021+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-03-06T16:03:53.021+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-03-06T16:03:53.022+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-03-06T16:03:53.022+0000] {spark_submit.py:641} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-03-06T16:03:53.023+0000] {spark_submit.py:641} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-06T16:03:53.023+0000] {spark_submit.py:641} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2025-03-06T16:03:53.024+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-03-06T16:03:53.024+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-03-06T16:03:53.024+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-06T16:03:53.025+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-06T16:03:53.025+0000] {spark_submit.py:641} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-06T16:03:53.089+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://172.18.0.4:7077...
[2025-03-06T16:03:53.130+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO TransportClientFactory: Successfully created connection to /172.18.0.4:7077 after 22 ms (0 ms spent in bootstraps)
[2025-03-06T16:03:53.223+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250306160353-0013
[2025-03-06T16:03:53.227+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306160353-0013/0 on worker-20250306150809-172.18.0.7-35707 (172.18.0.7:35707) with 2 core(s)
[2025-03-06T16:03:53.230+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306160353-0013/0 on hostPort 172.18.0.7:35707 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T16:03:53.231+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306160353-0013/1 on worker-20250306150810-172.18.0.10-33947 (172.18.0.10:33947) with 2 core(s)
[2025-03-06T16:03:53.232+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306160353-0013/1 on hostPort 172.18.0.10:33947 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T16:03:53.236+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 34339.
[2025-03-06T16:03:53.237+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO NettyBlockTransferService: Server created on e9dfa152b6ab:34339
[2025-03-06T16:03:53.239+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-03-06T16:03:53.245+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, e9dfa152b6ab, 34339, None)
[2025-03-06T16:03:53.252+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO BlockManagerMasterEndpoint: Registering block manager e9dfa152b6ab:34339 with 434.4 MiB RAM, BlockManagerId(driver, e9dfa152b6ab, 34339, None)
[2025-03-06T16:03:53.254+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, e9dfa152b6ab, 34339, None)
[2025-03-06T16:03:53.256+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, e9dfa152b6ab, 34339, None)
[2025-03-06T16:03:53.268+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250306160353-0013/1 is now RUNNING
[2025-03-06T16:03:53.269+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250306160353-0013/0 is now RUNNING
[2025-03-06T16:03:53.549+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-03-06T16:03:53.842+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-03-06T16:03:53.846+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:53 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2025-03-06T16:03:55.847+0000] {spark_submit.py:641} INFO - Traceback (most recent call last):
[2025-03-06T16:03:55.848+0000] {spark_submit.py:641} INFO - File "/opt/***/jobs/python/load_exchangeRateToDW.py", line 65, in <module>
[2025-03-06T16:03:55.853+0000] {spark_submit.py:641} INFO - .load()
[2025-03-06T16:03:55.854+0000] {spark_submit.py:641} INFO - ^^^^^^
[2025-03-06T16:03:55.855+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/streaming/readwriter.py", line 304, in load
[2025-03-06T16:03:55.860+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-03-06T16:03:55.866+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2025-03-06T16:03:55.880+0000] {spark_submit.py:641} INFO - pyspark.errors.exceptions.captured.AnalysisException: Failed to find data source: kafka. Please deploy the application as per the deployment section of Structured Streaming + Kafka Integration Guide.
[2025-03-06T16:03:56.020+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO SparkContext: Invoking stop() from shutdown hook
[2025-03-06T16:03:56.021+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-03-06T16:03:56.043+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO SparkUI: Stopped Spark web UI at http://e9dfa152b6ab:4040
[2025-03-06T16:03:56.051+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-03-06T16:03:56.058+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2025-03-06T16:03:56.084+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-03-06T16:03:56.096+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO MemoryStore: MemoryStore cleared
[2025-03-06T16:03:56.097+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO BlockManager: BlockManager stopped
[2025-03-06T16:03:56.104+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-03-06T16:03:56.107+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-03-06T16:03:56.117+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO SparkContext: Successfully stopped SparkContext
[2025-03-06T16:03:56.118+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO ShutdownHookManager: Shutdown hook called
[2025-03-06T16:03:56.118+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO ShutdownHookManager: Deleting directory /tmp/spark-cc767379-329b-485a-a732-d782f618dd54
[2025-03-06T16:03:56.122+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO ShutdownHookManager: Deleting directory /tmp/spark-0663f31e-5701-45fc-9d75-88545538b394
[2025-03-06T16:03:56.126+0000] {spark_submit.py:641} INFO - 25/03/06 16:03:56 INFO ShutdownHookManager: Deleting directory /tmp/spark-cc767379-329b-485a-a732-d782f618dd54/pyspark-78986be3-b341-4137-891a-f5988806bdf5
[2025-03-06T16:03:56.177+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.4:7077 --conf spark.jars=file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T16:03:56.181+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=spark_flow, task_id=load_exchangeRate_python_job, run_id=manual__2025-03-06T16:03:42.400789+00:00, execution_date=20250306T160342, start_date=20250306T160346, end_date=20250306T160356
[2025-03-06T16:03:56.216+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-03-06T16:03:56.217+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 49 for task load_exchangeRate_python_job (Cannot execute: spark-submit --master spark://172.18.0.4:7077 --conf spark.jars=file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.; 3462)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.4:7077 --conf spark.jars=file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T16:03:56.250+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-03-06T16:03:56.277+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-06T16:03:56.282+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
